homeassistant/components/zha/core/cluster_handlers/manufacturerspecific.py

```diff
@@ -410,26 +425,36 @@ class DanfossTRVChannel(ClusterHandler):
 
 
 @registries.CLUSTER_HANDLER_ONLY_CLUSTERS.register(
-    registries.DANFOSS_TRV_DIAGNOSTIC_CLUSTER
+    zigpy.zcl.clusters.hvac.UserInterface.cluster_id
 )
 @registries.ZIGBEE_CLUSTER_HANDLER_REGISTRY.register(
-    registries.DANFOSS_TRV_INTERFACE_CLUSTER
+    zigpy.zcl.clusters.hvac.UserInterface.cluster_id
 )
 class DanfossTRVInterfaceChannel(ClusterHandler):
     """Interface Channel class for the Danfoss TRV and derivatives."""
 
+    @classmethod
+    def matches(cls, cluster: zigpy.zcl.Cluster, endpoint: Endpoint) -> bool:
+        """Filter the cluster match for specific devices."""
+        return cluster.endpoint.device.manufacturer in DANFOSS_TRVS
```

Comment:
You might be able to use `cluster.endpoint.device.quirk_class` here.
However, https://github.com/home-assistant/core/pull/93268/files changed the "matches" to only check the "last two strings" when separated by a dot (`.`). So the code here should also check like that.

IMO, this "last two string getter" should be put in the base class though, so we don't do the same thing in multiple places.
Not sure if it should be done in this PR though.

---

But tbh, I don't like this whole "quirk class" matching system, as it can change during refactors and with now only taking "the last two strings", it's not even required that the manufacturer appears in that "quirk class" depending on how zha-quirks is structured for the manufacturer.

For example, I'd rather prefer something like a "quirk ID" (just a simple string that's set in each quirk that needs to be matched against in ZHA) that uniquely identifies a quirk — not something that can change like a file name/when refactoring). (But this needs to be discussed first and not done here)

Reply:
I agree that there should be a more maintainable way to match quirks. Although I do believe that the method now is better than my first attempt :). I'll do testing on device again to validate tomorrow.